wagtail_newsletter/static/wagtail_newsletter/js/wagtail_newsletter.js

```diff
@@ -65,53 +49,48 @@ window.wagtail.app.register("wn-panel",
         this.sendButtonProgress.loadingValue = false;
       }
     }
-
-    send() {
-      this.sendButtonProgress.activate();
-      this.sendSubmitTarget.click();
-    }
   }
 );
 
 
-window.wagtail.app.register("wn-test",
+window.wagtail.app.register("wn-send",
   class extends window.StimulusModule.Controller {
+    static targets = [
+      "message",
+    ];
+
     get dialog() {
       return this.application.getControllerForElementAndIdentifier(
         this.element.closest("[data-controller=w-dialog]"), "w-dialog"
       );
     }
 
-    submit() {
-      const address = this.element.querySelector("input[name=email]").value;
-      this.dispatch("submit", { detail: { address } });
-      this.dialog.hide();
+    show(event) {
+      const { name, member_count } = event.detail;
+      this.messageTarget.textContent = `Sending campaign to ${name}, with ${member_count} recipients.`;
+      this.dialog.show();
     }
   }
 );
 
 
-window.wagtail.app.register("wn-send",
+window.wagtail.app.register("wn-submit",
   class extends window.StimulusModule.Controller {
     static targets = [
-      "message",
-    ];
+      "button",
+    ]
 
-    get dialog() {
-      return this.application.getControllerForElementAndIdentifier(
-        this.element.closest("[data-controller=w-dialog]"), "w-dialog"
-      );
+    get button() {
+      return this.hasButtonTarget ? this.buttonTarget : this.element;
     }
 
-    show(event) {
-      const { name, member_count } = event.detail;
-      this.messageTarget.textContent = `Sending campaign to ${name}, with ${member_count} recipients.`;
-      this.dialog.show();
+    enter() {
+      setTimeout(() => this.button.click(), 0);
     }
 
-    submit() {
-      this.dispatch("submit");
-      this.dialog.hide();
+    sendEvent(event) {
+      const eventName = this.button.value;
```

Comment:
This might be non-obvious: the stimulus event name is picked up from the button's `value` attribute. So for a button defined like this:

```html
<button type="submit" name="newsletter-action" value="send_test_email">Send test email</button>
```

The component will dispatch an event named `wn-submit:send_test_email`.

This helps each user-visible non-submit button to catch the right event and trigger its spinner.

Reply:
I've actually included this as a comment in the code.